nix/global-node-packages/default.nix

```diff
@@ -0,0 +1,31 @@
+{ pkgs, nodejs }:
+
+let nodePackages = import ./output { inherit pkgs; inherit nodejs; };
+    realm-version = "5.14.0";
+    realm-core-src = pkgs.fetchurl (
+      if builtins.currentSystem == "x86_64-darwin" then {
+        url = "https://static.realm.io/downloads/core/realm-core-Release-v${realm-version}-Darwin-devel.tar.gz";
+        sha256 = "533157aecab9c3c504c12927c68a44fd0c4059bcff9b8e951bb2f023d680e8fc";
+      } else {
+        url = "https://static.realm.io/downloads/core/realm-core-Release-v${realm-version}-Linux-devel.tar.gz";
+        sha256 = "1lxsyfhqxkb6zkvb88yq7lzdjj5iylhk6x7hkgpmh6x5mkrp0m5h";
+      }
+    );
+    realm-dest-dir = if builtins.currentSystem == "x86_64-darwin" then
+      "$out/lib/node_modules/realm/vendor/realm-darwin-x64" else
+      "$out/lib/node_modules/realm/vendor/realm-linux-x64";
+in nodePackages // {
+  "realm-2.26.0" = nodePackages."realm-2.26.0".override(oldAttrs: {
+    buildInputs = oldAttrs.buildInputs ++ [ pkgs.nodePackages.node-pre-gyp ];
+    bypassCache = true;
+    reconstructLock = true;
+    preRebuild = ''
+      # Do not attempt to do any http calls!
+      # delete the last callback at the bottom of download-realm.js
+      head -265 $out/lib/node_modules/realm/scripts/download-realm.js > \
```

Comment:
This doesn't look very safe compared to `substituteInPlace`. Why this choice?

Reply:
yes this isn't safe and could need a change when changing realm versions. I got syntax error for the dot in then with substituteInPlace, even if I returned true.

Reply:
alternatively, find the last expression and use that line number instead of hardcoding?

Reply:
> yes this isn't safe and could need a change when changing realm versions. I got syntax error for the dot in then with substituteInPlace, even if I returned true.

Newbie question: what do you mean with "the dot"?

Since we now have 2 separate expressions for mobile and desktop, we could also make a .patch file for each.